src/services/web3/useVueWeb3.ts

```diff
@@ -0,0 +1,104 @@
+import { getProfile } from '@/lib/utils/profile';
+import axios from 'axios';
+import { computed, inject, reactive, ref, watch } from 'vue';
+import { useQuery } from 'vue-query';
+import { Web3Plugin, Web3ProviderSymbol } from './web3.plugin';
+import { Web3Provider } from '@ethersproject/providers';
+import QUERY_KEYS from '@/constants/queryKeys';
+import ConfigService from '../config/config.service';
+
+const isWalletSelectVisible = ref(false);
+
+export default function useVueWeb3() {
+  const {
+    connectWallet,
+    account,
+    chainId,
+    disconnectWallet,
+    connector,
+    provider,
+    walletState
+  } = inject(Web3ProviderSymbol) as Web3Plugin;
+  const appConfig = new ConfigService();
+
+  const userNetworkConfig = computed(() =>
+    appConfig.getNetworkConfig(String(chainId.value))
+  );
+  const appNetworkConfig = appConfig.network;
+
+  // if the account ref has changed, we know that
+  // the user has successfully connected a wallet
+  watch(account, () => {
+    toggleWalletSelectModal(false);
+  });
+
+  // METHODS
+  const toggleWalletSelectModal = (value: boolean) => {
+    if (value !== undefined && typeof value === 'boolean') {
+      isWalletSelectVisible.value = value;
+      return;
+    }
+    isWalletSelectVisible.value = !isWalletSelectVisible.value;
+  };
+
+  const isWalletReady = computed(() => walletState.value === 'connected');
+
+  const canLoadProfile = computed(
+    () => account.value !== '' && userNetworkConfig.value.chainId !== 0
+  );
+
+  const getProvider = () => new Web3Provider(provider.value as any);
+
+  // TODO separate this out?
+  const { isLoading: isLoadingProfile, data: profile } = useQuery(
+    QUERY_KEYS.Account.Profile(account, userNetworkConfig),
+    () => getProfile(account.value, String(userNetworkConfig.value.chainId)),
+    reactive({
+      enabled: canLoadProfile
+    })
+  );
```

Comment:
I'd quite like to see this separated out. In the format we have the other queries. I know it's a really simple one, but if you're not familiar with reactQuery it's actually quite hard to parse and understand what's happening.

Also, could you adjust `getProfile` to not fetch the 3box data. We basically only want to fetch the ens name now.

Reply:
No worries.

Reply:
Leaving this here actually, conversation around too much abstraction. It is a pretty simply query with not much ancillary compute happening, so will leave here. We can revisit to separate it out if it becomes a big boy